src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMapping.java

```diff
@@ -290,15 +289,16 @@ private void validateVirtualPhysicalCompatibility() {
     void validateSecondaryIndexes(DynamoTableDescription virtualTable,
                                   DynamoTableDescription physicalTable,
                                   DynamoSecondaryIndexMapper secondaryIndexMapper) {
-        virtualTable.getSis().stream().map(virtualSi -> {
+        checkArgument(virtualTable.getSis().size() == virtualTable.getSis().stream().map(virtualSi -> {
```

Comment:
if `virtualTable.getSis()` is at all expensive, consider (calling the method once and) saving its result into a local variable instead of calling the method twice

Reply:
That call isn't expensive and it's expected that `TableMapping`s will be cached.  However, it's probably more consistent and coherent to make it a variable so I made the change regardless in a8a649270fe3bec1b4903e51994bc08f2bd7672d.